components/plan-build/bin/hab-plan-build.sh

```diff
@@ -1144,24 +1144,24 @@ _populate_dependency_arrays() {
   # Build `${pkg_all_deps_resolved[@]}` containing all direct build and run
   # dependencies. The build dependencies appear before the run dependencies.
   pkg_all_deps_resolved=(
-    "${pkg_deps_resolved[@]}"
     "${pkg_build_deps_resolved[@]}"
+    "${pkg_deps_resolved[@]}"
   )
 
   # Build an ordered set of all build and run dependencies (direct and
   # transitive). The order is important as this gets used when setting the
   # `$PATH` ordering in the build environment. To give priority to direct
   # dependencies over transitive ones the order of packages is the following:
   #
-  # 1. All direct run dependencies
-  # 2. All direct build dependencies
-  # 3. All unique transitive run dependencies that aren't already added
-  # 4. All unique transitive build dependencies that aren't already added
+  # 1. All direct build dependencies
+  # 2. All direct run dependencies
+  # 3. All unique transitive build dependencies that aren't already added
+  # 4. All unique transitive run dependencies that aren't already added
   pkg_all_tdeps_resolved=(
-    "${pkg_deps_resolved[@]}"
     "${pkg_build_deps_resolved[@]}"
+    "${pkg_deps_resolved[@]}"
   )
-  for dep in "${pkg_tdeps_resolved[@]}" "${pkg_build_tdeps_resolved[@]}"; do
+  for dep in "${pkg_build_tdeps_resolved[@]}" "${pkg_tdeps_resolved[@]}"; do
```

Comment:
Is it possible to have some tests for this behavior? It's hard to determine from reading the code if this does the right thing (or if this is reversed by the code adding these to the `$PATH`)

😭